pkg/k8s/service.go

```diff
@@ -351,6 +351,10 @@ func (s *Service) DeepEqual(other *Service) bool {
 		return false
 	}
 
+	if s.Shared != other.Shared || s.IncludeExternal != other.IncludeExternal {
+		return false
+	}
+
```

Comment:
You can remove the `// +deepequal-gen=false` from both fields and regenerate the deepequal code again.

Reply:
thanks :100:

Reply:
originally, I think we didn't use the DeepEqual from generated code, but just a lightweitght DeepEqual here, however, I think we should consider these two fields for general equality check.

Forced push with below diff

```diff
diff --git a/pkg/k8s/service.go b/pkg/k8s/service.go
index 72b211c1a..18aa7d07f 100644
--- a/pkg/k8s/service.go
+++ b/pkg/k8s/service.go
@@ -286,11 +286,9 @@ type Service struct {
 
 	// IncludeExternal is true when external endpoints from other clusters
 	// should be included
-	// +deepequal-gen=false
 	IncludeExternal bool
 
 	// Shared is true when the service should be exposed/shared to other clusters
-	// +deepequal-gen=false
 	Shared bool
 
 	// TrafficPolicy controls how backends are selected. If set to "Local", only
diff --git a/pkg/k8s/zz_generated.deepequal.go b/pkg/k8s/zz_generated.deepequal.go
index 09625f22c..7f6b20abb 100644
--- a/pkg/k8s/zz_generated.deepequal.go
+++ b/pkg/k8s/zz_generated.deepequal.go
@@ -158,6 +158,12 @@ func (in *Service) deepEqual(other *Service) bool {
 	if in.IsHeadless != other.IsHeadless {
 		return false
 	}
+	if in.IncludeExternal != other.IncludeExternal {
+		return false
+	}
+	if in.Shared != other.Shared {
+		return false
+	}
 	if in.TrafficPolicy != other.TrafficPolicy {
 		return false
 	}

```